ros_ws/src/get_user_request.py

```diff
@@ -0,0 +1,141 @@
+import os
+import langchain
+
+from langchain.llms import OpenAI
+from langchain import PromptTemplate
+from langchain import schema
+from langchain.prompts import (ChatPromptTemplate,PromptTemplate,SystemMessagePromptTemplate, HumanMessagePromptTemplate)
+from langchain.chat_models import ChatOpenAI
+from langchain.chains import LLMChain
+from langchain.schema import HumanMessage, SystemMessage, AIMessage
+api_key=os.environ["OPENAI_API_KEY"]
+model_to_use="gpt-4-0613"
+#model_to_use="text-davinci-003"
+
+
+def Refine(request):
+    chat = ChatOpenAI(model_name=model_to_use)
+
+    sys_msg=SystemMessage(content="""You are a home assistant robot that can retrieve objects for your user. 
```

Comment:
Long blocks of text generally aren't good in sets of code like this as it can add a lot of length; I typically put prompts into their own files or imports. This is on the edge in length and can go either way so I'll leave it to your preference.